### DISCUSSION TITLE
Question on how to display selected faces of few cells

### DISCUSSION THREAD

**Discussion by RadheChandu** [usnistgov/fipy#954] [Q&A]
Hello fipyTeam,

I am working on a problem in which I need to apply flux boundary conditions on few internal cells.  I am able to select the cells. But now I want to select their exterior faces and also would like to see in Viewer, if I did select the correct cells and Faces. Can someone please help me with this. 

This is the sample code. 

`
from fipy import CellVariable, Grid3D, Viewer, numerix, FaceVariable

mesh = Grid3D(.1,.1,.1, 40, 40, 10)

mesh = mesh + ((-2,), (-2,), (-1,))

alp = CellVariable(mesh, value = 0.)

X, Y, Z = mesh.cellCenters

radius = 0.5

selected = X**2 + Y**2 < radius**2

alp.setValue(1., where = selected)

viewer = Viewer(vars = alp)`

I was able to check that I selected the correct cells. but how to visualize the selected faces. 

I tried something like this.. 

`
allFaces = mesh.cellFaceIDs.data

selectedCellFaces = numerix.unique(allFaces[:, selected]).flatten()
unselectedCellFaces = numerix.unique(allFaces[:, numerix.invert(selected)]).flatten()

selectedExteriorFaces = numerix.intersect1d(selectedCellFaces, unselectedCellFaces)

SelectFaces = numerix.zeros(mesh.numberOfFaces, dtype = bool)

SelectFaces[selectedExteriorFaces] = True

beta = FaceVariable(mesh, value = 0.)

beta.setValue(10, where = SelectFaces)

viewer = Viewer(vars = beta)`

But its showing cellcenters of mesh and face values as points. Is there any better way of doing this?

![image](https://github.com/usnistgov/fipy/assets/139492124/492d9448-b730-4314-a7b5-953f1e1898c9)

**Comment by guyer** [comment]
I'm afraid we don't have anything that will improve the visualization. Visualization of `FaceVariables` is just not something we've prioritized. For that matter, we haven't spent a ton of time developing general tools for 3D visualization, because it's just so problem-specific. On the bright side, the full power of mayavi is at your disposal with a custom [`MayaviDaemon`](https://www.ctcms.nist.gov/fipy/fipy/generated/fipy.viewers.mayaviViewer.html#fipy.viewers.mayaviViewer.mayaviDaemon.MayaviDaemon), as illustrated in [`examples.cahnHilliard.sphere`](https://www.ctcms.nist.gov/fipy/examples/cahnHilliard/generated/examples.cahnHilliard.sphere.html). You're on your own figuring out what to tell mayavi, though.

My recommendation, though, is to use Gmsh to define the boundary explicitly, akin to what's shown in the [`Gmsh2D` documentation](https://www.ctcms.nist.gov/fipy/fipy/generated/fipy.meshes.html#fipy.meshes.gmshMesh.Gmsh2D). Then you can be confident that the boundary faces are where you want them and can access them easily with `my_mesh.physicalFaces["my_boundary_label"]`.

**Reply by RadheChandu** [reply]
Thank you for your reply.. I will look at the sources and will try to figure out something..
